plutus-chain-index/src/Plutus/ChainIndex/DbStore.hs

```diff
@@ -110,33 +110,58 @@ instance Table AddressRowT where
     data PrimaryKey AddressRowT f = AddressRowId (Columnar f ByteString) (Columnar f ByteString) deriving (Generic, Beamable)
     primaryKey (AddressRow c o) = AddressRowId c o
 
-data UtxoRowT f = UtxoRow
-    { _utxoRowSlot        :: Columnar f Word64 -- In Plutus Slot is Integer, but in the Cardano API it is Word64, so this is safe
-    , _utxoRowBlockId     :: Columnar f ByteString
-    , _utxoRowBlockNumber :: Columnar f Word64
-    , _utxoRowBalance     :: Columnar f ByteString
+data TipRowT f = TipRow
+    { _tipRowSlot        :: Columnar f Word64 -- In Plutus Slot is Integer, but in the Cardano API it is Word64, so this is safe
+    , _tipRowBlockId     :: Columnar f ByteString
+    , _tipRowBlockNumber :: Columnar f Word64
     } deriving (Generic, Beamable)
 
-type UtxoRow = UtxoRowT Identity
+type TipRow = TipRowT Identity
 
-instance Table UtxoRowT where
-    data PrimaryKey UtxoRowT f = UtxoRowId (Columnar f Word64) deriving (Generic, Beamable)
-    primaryKey = UtxoRowId . _utxoRowSlot
+instance Table TipRowT where
+    data PrimaryKey TipRowT f = TipRowId { unTipRowId :: Columnar f Word64 } deriving (Generic, Beamable)
+    primaryKey = TipRowId . _tipRowSlot
+
+data UnspentOutputRowT f = UnspentOutputRow
+    { _unspentOutputRowTip    :: PrimaryKey TipRowT f
```

Comment:
Maybe add a comment saying if this is the tip which produced the utxo or the actual current tip? Not sure I know which one it is.

Reply:
Good point, while explaining I realised it is a bit more complicated than I thought.